ffs/filcold/filcold.go

```diff
@@ -284,35 +287,38 @@ Loop:
 		case <-time.After(timeout):
 			msg := fmt.Sprintf("DealID %d with miner %s tracking timed out after waiting for %.0f hours.", last.DealID, last.Miner, timeout.Hours())
 			fc.l.Log(ctx, msg)
-			return ffs.FilStorage{}, ffs.DealError{ProposalCid: proposal, Message: msg}
+			return ffs.FilStorage{}, ffs.DealError{ProposalCid: proposal, Miner: last.Miner, Message: msg}
 		case di, ok := <-chDi:
 			if !ok {
 				break Loop
 			}
 			last = di
-			switch di.StateID {
+			proposal := ffs.FilStorage{
+				ProposalCid:     di.ProposalCid,
+				Status:          di.Status,
+				PieceCid:        di.PieceCID,
+				Duration:        int64(di.Duration),
+				Miner:           di.Miner,
+				ActivationEpoch: di.ActivationEpoch,
+				StartEpoch:      di.StartEpoch,
+				EpochPrice:      di.PricePerEpoch,
+			}
+			dealUpdates <- proposal
```

Comment:
Every time we get an update from the deals module, we create a `FilStorage` with the current state and send it on the chan.

Reply:
Maybe is better to change this last line to:
```
select {
   case dealUpdates <- proposal:
   default:
        log.Warnf("slow receiver for deal updates for %s", c)
}
```
If the other side is slow or abandoned the channel, it's better to skip the notification and move on than be blocked here.

Reply:
Cool, I've wondered how this "slow receiver" pattern works... How does it work? What does it mean to be slow and how is that detected?

Reply:
When you do `dealUpdates <- proposal`, it will block until the receiver side reads the channel.
If the receiver is too slow reading the channel, this is might be a problem since you will be blocked at the sender side.

That `select` just avoids blocking and skips the push if the receiver isn't listening... so to never block.

Reply:
Maybe in this case we can assume blocking is OK since the receiver should be fast enough or never abandon reading the channel. It's usually a judgment call.